druid/src/main/java/org/apache/calcite/adapter/druid/DruidQuery.java

```diff
@@ -813,6 +815,29 @@ protected QuerySpec getQuery(RelDataType rowType, RexNode filter, List<RexNode>
         generator.writeEndObject();
         break;
 
+      case SCAN:
+        generator.writeStartObject();
+
+        generator.writeStringField("queryType", "scan");
+        generator.writeStringField("dataSource", druidTable.dataSource);
+        writeField(generator, "intervals", intervals);
+        writeFieldIf(generator, "filter", jsonFilter);
+        writeField(generator, "columns",
+            Lists.transform(fieldNames, new Function<String, String>() {
```

Comment:
can we use native java 8 function.

Reply:
@nishantmonu51 this still open

Reply:
Calcite is shortly dropping JDK 7 support so we could in theory start using lambdas etc. but Guava is extensively used so let's continue to use it for now.

Reply:
Thanks, thought Calcite is Java 8 and higher.